application/F3DStarter.cxx

```diff
@@ -180,11 +180,15 @@ int F3DStarter::Start(int argc, char** argv)
       [this](std::vector<std::string> filesVec) -> bool
       {
         this->Internals->Engine->getInteractor().stopAnimation();
+        int index = -1;
         for (std::string file : filesVec)
         {
-          this->AddFile(fs::path(file));
+          index = this->AddFile(fs::path(file));
+        }
+        if (index > -1)
+        {
+          this->LoadFile(index);
```

Comment:
what happends if you drop the folder containing the currently loaded file ?

If I understand well it loads the last file of the folder. I think that it is fine, just checking.

Reply:
It would display whatever was "added" last, according to whatever the traversing/recursing order is, I assume. I haven't really looked into the recursive cases tbh.

Reply:
as long as it is not broken and file are actually added, fine by me.

Reply:
It would need to be looked into deeper to be done properly right. Not sure I'm up to dealing with what I expect to find down there though (eg. it will shove a bunch of possibly random files in the list on add, and we'd need to handle the unreadable files to skip them and remove them from the list on load).

Reply:
I just tested `master` it with this:

`./bin/f3d ../src/testing/data/multi/suzanne.obj`

Then dropped multi into it, zsuzanne.stl was loaded.
So current behavior is as described, and you did not change that as far as I can see, so this is perfectly fine :)


> we'd need to handle the unreadable files to skip them and remove them from the list on load).

Not needed, you can drop an unreadable file, it will just show up as an empty "UNSUPPORTED" file. There is already tests for that.

Reply:
> Not needed, you can drop an unreadable file, it will just show up as an empty "UNSUPPORTED" file. There is already tests for that.

Good to know it's a feature and not a bug that needs fixing. Scenarios that may be worth considering however:
1. I open an `.stl` I saved in my `~/Download` folder full of random stuff, I press down to see what other 3d files I have in there, I am now scrolling through a possibly high percentage of "[UNSUPPORTED]" memes, cat pictures, movies...
2. I open a model in my home directory I press down by accident with my fat fingers, F3D it now off rummaging through the depth of my `~/`, hidden folders and all...

Reply:
Correct, this feature may need some improvements, but this is completely outside of the scope of this PR. Thanks for checking.